### DISCUSSION TITLE
Markdown support in quill

### DISCUSSION THREAD

**Discussion by amanpreetSO** [slab/quill#4325] [Q&A]
hello Team, 
I have just integrated the quill into my angular project but this seems like i found a blocker. I am not able to find the way to parse the markdown to html and html to markdown in quill editor.
I found this npm package https://www.npmjs.com/package/quilljs-markdown#markdown-for-quill-editor
but this only works if you paste the markdown in editor but if markdown is already saved in your DB then its not rending in HTML and when I save the content then it again give the HTML not the markdown. I want to save the editor content in markdown in DB instead of HTML. How can I get it. I am using ngx-quill 23.0.2 version.

**Comment by nibl** [comment]
Use getSemanticHTML(), then convert to Markdown using a library in whatever programming language you are using, e.g. https://github.com/mixmark-io/turndown in  Javascript.

But you will still need to save in Quill's Delta format because that is the only way to load/import data.

**Reply by rxliuli** [reply]
Not entirely, HTML cannot be losslessly converted to Quill Delta format, for example, the `<hr/>` tag requires custom conversion functionality. On the other hand, I am submitting a [PR](https://github.com/frysztak/markdown-to-quill-delta/pull/7) for another package [markdown-to-quill-delta](https://github.com/frysztak/markdown-to-quill-delta) to fix some issues to correctly handle markdown to delta format.

**Comment by trymeouteh** [comment]
https://github.com/slab/quill/discussions/4584
